File: backend/server/datastore/commands/account/account_profile.py
from django.conf import settings

from server.apps.account.models import AccountProfile


class AccountProfileCommand:
    @classmethod
    def anonymize(cls, account_profile: AccountProfile) -> AccountProfile:
        account_profile.first_name = settings.ANONYMIZED_FIRST_NAME
        account_profile.last_name = settings.ANONYMIZED_LAST_NAME
        account_profile.phone_number = None
        account_profile.avatar = None
        account_profile.id_card = None

        account_profile.save()

        return account_profile

File: mobile/lib/features/account_settings/domain/entities/account_profile.dart
import 'package:equatable/equatable.dart';

class AccountProfile extends Equatable {
  final String firstName;
  final String lastName;
  final String? phoneNumber;
  final String? avatar;
  final String? idCard;

  const AccountProfile({
    required this.firstName,
    required this.lastName,
    this.phoneNumber,
    this.avatar,
    this.idCard,
  });

  copyWith({
    String? firstName,
    String? lastName,
    String? phoneNumber,
    String? avatar,
    String? idCard,
  }) =>
      AccountProfile(
        firstName: firstName ?? this.firstName,
        lastName: lastName ?? this.lastName,
        phoneNumber: phoneNumber ?? this.phoneNumber,
        avatar: avatar ?? this.avatar,
        idCard: idCard ?? this.idCard,
      );

  @override
  List<Object?> get props => [
        firstName,
        lastName,
        phoneNumber,
        avatar,
        idCard,
      ];
}
